packages/account/tests/account.e2e.spec.ts

```diff
@@ -90,20 +86,73 @@ describe("Account Tests", () => {
     };
 
     const balance = (await checkBalance(publicClient, recipient, nftAddress)) as bigint;
-    const partialUserOp = await smartWallet.buildUserOp([transaction]);
 
-    const paymasterData = await paymaster.getPaymasterAndData(partialUserOp, {
-      mode: PaymasterMode.SPONSORED,
+    const maticBalanceBefore = publicClient.getBalance({ address: await smartWallet.getAddress() });
+
+    const response = await smartWallet.sendTransaction(transaction, { paymasterServiceData: { mode: PaymasterMode.SPONSORED } });
+
+    const userOpReceipt = await response.wait(3);
+    expect(userOpReceipt.userOpHash).toBeTruthy();
+    expect(userOpReceipt.success).toBe("true");
+
+    const maticBalanceAfter = publicClient.getBalance({ address: await smartWallet.getAddress() });
+
+    expect((await maticBalanceAfter).toString()).toEqual((await maticBalanceBefore).toString());
+
+    const newBalance = (await checkBalance(publicClient, recipient, nftAddress)) as bigint;
+
+    expect(newBalance - balance).toBe(1n);
+  }, 60000);
+
+  it("Should mint an NFT on Mumbai and pay with ERC20", async () => {
+    const nftAddress: Hex = "0x1758f42Af7026fBbB559Dc60EcE0De3ef81f665e";
+    const {
+      whale: { viemWallet: signer, publicAddress: recipient },
+      bundlerUrl,
+      biconomyPaymasterApiKey,
+      publicClient,
+    } = mumbai;
+
+    const smartWallet = await createSmartWalletClient({
+      signer,
+      bundlerUrl,
+      biconomyPaymasterApiKey,
+    });
+
+    const encodedCall = encodeFunctionData({
+      abi: parseAbi(["function safeMint(address _to)"]),
+      functionName: "safeMint",
+      args: [recipient],
+    });
+
+    const transaction = {
+      to: nftAddress, // NFT address
+      data: encodedCall,
+      value: 0,
```

Comment:
If you backmerge you should be able to remove the value here in a typesafe manner.

Reply:
Did it work? This will end up in all our documentation if it doesn't. It's worth testing